### PR TITLE
agent: set content type header explicitly

### PR DESCRIPTION
**Pull request by pkazmierczak**
This PR addresses an XSS vulnerability where Nomad agents wouldn't explicitly set content type headers for error responses. 

Internal ref: https://hashicorp.atlassian.net/browse/SECVULN-14567

### CODE REVIEW

File: command/agent/http.go [hunk 1]
```diff
@@ -745,6 +748,7 @@ func (s *HTTPServer) wrap(handler func(resp http.ResponseWriter, req *http.Reque
 
 			resp.WriteHeader(code)
 			resp.Write([]byte(errMsg))
+			resp.Header().Set(contentTypeHeader, plainContentType)
```
Comment (schmichael) [inline command/agent/http.go:751]:
```suggestion
			resp.Header().Set(contentTypeHeader, plainContentType)
			resp.WriteHeader(code)
			resp.Write([]byte(errMsg))
```

I think setting the header needs to come before the other 2 response calls.

Comment (pkazmierczak) [reply command/agent/http.go:751]: oh, does it matter? I didn't realize. I'll update the rest of occurrences.

Comment (dduzgun-security) [reply command/agent/http.go:751]: setting the headers first ensures that the proper metadata / content-type can be sent back which reduces the risk possible wrong encoding or other response manipulation

Comment (pkazmierczak) [reply command/agent/http.go:751]: ah `WriteHeader` writes the header immediately to the response object 🤦‍♂️ 
"it's all coming back to me now"

Comment (pkazmierczak) [reply command/agent/http.go]: fixed in 7410b7312efcb2eac422615d8be138cd2ca3606d

**Review by schmichael — APPROVED** [review]

**Review by dduzgun-security — APPROVED** [review]

**Pull request merged**